sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventDataBatch.cs

```diff
@@ -173,17 +181,17 @@ public bool TryAdd(EventData eventData)
                 AssertNotLocked();
 
                 var messageScopeCreated = false;
-                var identifier = default(string);
+                var traceparent = default(string);
 
                 try
                 {
-                    (messageScopeCreated, identifier) = EventDataInstrumentation.InstrumentEvent(eventData, FullyQualifiedNamespace, EventHubName);
+                    ClientDiagnostics.InstrumentMessage(eventData.Properties, DiagnosticProperty.EventActivityName, out messageScopeCreated, out traceparent, out var tracestate);
```

Comment:
Why all the out params?   I find that much less clear than deconstructing a tuple on return.

Reply:
I thought it was more appropriate given that not all callers care about the return value. The out params represent extra info that you can get back as opposed to being the primary reason for calling the method. That said, I can change it back if you prefer.